File: src/NotificationFactory.java
import javax.swing.plaf.BorderUIResource;

public class NotificationFactory {
    public Notification createNotification(String channel){

        if (channel==null || channel.isEmpty())
            return null;

        return switch (channel) {
            case "SMS" -> new SMSNotification();
            case "EMAIL" -> new EmailNotification();
            case "PUSH" -> new PushNotification();
            default -> throw new IllegalArgumentException("Unknown Channel" + channel);
        };

    }
}

File: src/NotificationFactory.kt
import javax.swing.plaf.BorderUIResource;

object NotificationFactory {
    fun createNotification(channel : String) : Notification{

        return when (channel) {
             "SMS" -> SMSNotification()
             "EMAIL" -> EmailNotification()
             "PUSH" -> PushNotification()
            else -> {
                throw IllegalArgumentException("Unknown Channel$channel")
            }
        }

    }
}
